akka-contrib/src/main/scala/akka/contrib/pattern/ClusterSingletonManager.scala

```diff
@@ -191,8 +191,9 @@ object ClusterSingletonManager {
 
       def handleInitial(state: CurrentClusterState): Unit = {
         membersByAge = immutable.SortedSet.empty(ageOrdering) ++ state.members.filter(m ⇒
-          m.status == MemberStatus.Up && matchingRole(m))
-        val initial = InitialOldestState(membersByAge.headOption.map(_.address), membersByAge.size)
+          (m.status == MemberStatus.Up || m.status == MemberStatus.Leaving) && matchingRole(m))
+        val safeToBeOldest = !state.members.exists { m ⇒ (m.status == MemberStatus.Down || m.status == MemberStatus.Exiting) }
```

Comment:
Is this to guard against the previously oldest being in the process of leaving? In that case, would it make sense to compute the previously oldest?